src/main/scala/mesosphere/marathon/api/v2/TasksResource.scala

```diff
@@ -142,13 +160,18 @@ class TasksResource @Inject() (
       }))
     }
 
-    val tasksByAppId: Map[PathId, Seq[Instance]] = tasksToAppId.view
-      .flatMap { case (taskId, appId) => instanceTracker.instancesBySpecSync.instance(Task.Id(taskId).instanceId) }
-      .groupBy { instance => instance.instanceId.runSpecId }
-      .map { case (appId, instances) => appId -> instances.to[Seq] }(collection.breakOut)
-
-    if (scale) scaleAppWithKill(tasksByAppId)
-    else killTasks(tasksByAppId)
+    val futureResponse = async {
+      val maybeInstances = await(Future.sequence(tasksIdToAppId.view
+        .map { case (taskId, _) => instanceTracker.instancesBySpec.map(_.instance(taskId)) }))
+      val tasksByAppId: Map[PathId, Seq[Instance]] = maybeInstances.flatten
+        .groupBy { instance => instance.instanceId.runSpecId }
+        .map { case (appId, instances) => appId -> instances.to[Seq] }(collection.breakOut)
```

Comment:
`Iterable[Instance]` I will add type definition to maybeInstances so it will be easier to decode manually.